dsl/src/test/java/org/mule/test/extension/dsl/OcsArtifactDeclarationSerializerTestCase.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.extension.dsl;
+
+import static java.util.Arrays.asList;
+import static org.mule.runtime.app.declaration.api.fluent.ElementDeclarer.newArtifact;
+import static org.mule.runtime.app.declaration.api.fluent.ElementDeclarer.newParameterGroup;
+import static org.mule.runtime.extension.internal.ocs.OCSConstants.OCS_ENABLED;
+
+import org.mule.runtime.app.declaration.api.ArtifactDeclaration;
+import org.mule.runtime.app.declaration.api.fluent.ElementDeclarer;
+
+import java.util.Collection;
+
+import org.junit.AfterClass;
+import org.junit.runners.Parameterized;
+
+public class OcsArtifactDeclarationSerializerTestCase extends ArtifactDeclarationSerializerTestCase {
+
+  private static String ocsPropertyOldValue;
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+    ocsPropertyOldValue = System.setProperty(OCS_ENABLED, "true");
```

Comment:
Can't the utility method created on the other PR be used here?

Reply:
Not really because the property needs to be set not only to create the artifact but also during the test suite initialisation when extensions are discovered

Reply:
Would that be covered if we added a SystemProperty ClassRule?

Reply:
Also tried it and it didn't work